pkg/plugin/driver/job/job.go

```diff
@@ -196,6 +214,30 @@ func (p *Plugin) listOptions() metav1.ListOptions {
 	}
 }
 
+func (p *Plugin) getSecretName() string {
+	return fmt.Sprintf("job-%s-%s", p.GetName(), p.SessionID)
+}
+
+func (p *Plugin) createClientSecret(client kubernetes.Interface, key crypto.PrivateKey) error {
```

Comment:
This looks identical to the other plugin's implementation. Not necessarily for this PR but maybe we should think about common functions between plugins? plugin util? 😂

Reply:
I also noticed there is a fair amount of overlap that we could reduce over time.